File: LibraryApi/Models/Book.cs
namespace LibraryApi.Models;

public class Book
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public int Rating { get; set; }
    public string? Comments { get; set; }
    public string? Isbn { get; set; }
}

File: src/types/book.ts
// book interface
export interface Book {
  id: number
  title: string
  author: string
  rating: number
  comments?: string | null
  isbn: string
}

// book create interface
export interface BookCreate {
  title: string
  author: string
  rating: number
  comments?: string | null
  isbn: string
}

// book update interface
export interface BookUpdate {
  id: number
  rating: number
  comments?: string | null
}
